### PR TITLE
Added verification of projects from Management Portal for study page

### LINKED ISSUE #17: Verify Study Pages for Existing Management Portal Projects
There should be a way to verify that the study pages a user is trying to access already have an associated MP project. This can be achieved by utilizing the public endpoint used for management portal projects.

### PR DESCRIPTION
**Pull request by this-Aditya**
Fixes #17

### CODE REVIEW

**Review by yatharthranjan — COMMENTED** [review]
Looks good. just a couple of minor comments.

File: pages/study.tsx [hunk 1]
```diff
@@ -77,17 +89,31 @@ const StudyInfo: React.FC<any> = ({ questions }) => {
...
+    console.log(fetchedProjects)
+    const mpProjectNames: string[] = fetchedProjects.map((project: Project) => project.projectName)
+    if (!mpProjectNames.includes(projectId)) {
+      console.log("No project found")
+      return {
+        props: {
+          projectExists: false
+        }
+      };
+    }
```
[the remark this answers is not in the document]
Comment (this-Aditya) [reply pages/study.tsx:103]: Nice point, I will add the error in the UI similar to that one.

File: pages/study.tsx [hunk 2]
```diff
@@ -77,17 +105,53 @@ const StudyInfo: React.FC<any> = ({ questions }) => {
...
-      props: {
-        definitions: studyDefinitions,
+      if (studyDefinitions == undefined) return {props: {projectExists: true}}
+
+      return {
+        props: {
+          definitions: studyDefinitions,
+          projectExists: true
+        }
+      }
```
Comment (mpgxvii) [inline pages/study.tsx:131]:
```suggestion
return studyDefinitions === undefined 
  ? { props: { projectExists: true } } 
  : { props: { definitions: studyDefinitions, projectExists: true } }
```

File: pages/study.tsx [hunk 3]
```diff
@@ -77,17 +105,53 @@ const StudyInfo: React.FC<any> = ({ questions }) => {
 export const getServerSideProps: GetServerSideProps = async (context) => {
   const {projectId} = context.query
   if (typeof projectId === "string") {
-    const studyDefinitions: string | undefined = await githubService.initiateFetch(projectId,
-        REMOTE_DEFINITIONS_CONFIG.STUDY_INFO_DEFINITION_FILE_NAME_CONTENT, REMOTE_DEFINITIONS_CONFIG.STUDY_INFO_VERSION)
+    try {
+      const fetchedProjects: Project[] = await fetchProjectsFromMp()
+      console.log(fetchedProjects)
```
Comment (mpgxvii) [inline pages/study.tsx:110]: Maybe we can remove this log?

**Review by mpgxvii — APPROVED** [review]
LGTM! Thanks for this! Also, would it be possible to run Prettier? `npm run format`

**Comment by this-Aditya** [comment]
Merging this PR because running Prettier also formatted the files for the `fetch-definitions` branch. Merging these changes will provide a cleaner structure for review in that PR.

**Pull request merged**